dashboard/app/helpers/aichat_safety_helper.rb

```diff
@@ -0,0 +1,57 @@
+# Provides functionality to detect toxicity in user input and model output used in the AI Chat Lab.
+# Uses various services to check for profanity and toxicity based on DCDO settings.
+module AichatSafetyHelper
+  class ToxicityDetector
+    DEFAULT_TOXICITY_THRESHOLD_USER_INPUT = 0.2
+    DEFAULT_TOXICITY_THRESHOLD_MODEL_OUTPUT = 0.6
+
+    # Checks for toxicity in the given text using various services, determined by DCDO settings.
+    # Returns {text: input (string), blocked_by: serviced that detected toxicity (string), details: filtering details (hash)}
+    def find_toxicity(role, text, locale)
+      if blocklist_enabled?(role)
+        text.split.each do |word|
+          return {text: text, blocked_by: 'blocklist', details: {blocked_word: word}} if profane_word_blocklist.include? word
+        end
+      end
+
+      if webpurify_enabled?(role)
+        profanity = ShareFiltering.find_profanity_failure(text, locale)
```

Comment:
Related to comment above, this finds first occurrence of blocked word. We could call on `ProfanityFilter.flind_potential_profanities` and get the list of blocked words. May be interesting compare/contrast if both the custom block list and WebPurify are enabled.

Reply:
Ah interesting, didn't realize there was a difference - will check with Sam about this

Reply:
Thanks for checking! I see she said it's fine to stop after first blocked word. https://codedotorg.slack.com/archives/C06FELVTV0Q/p1726750801949449